x-pack/plugins/infra/public/utils/operation_status.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import last from 'lodash/fp/last';
+import { oc } from 'ts-optchain';
+
+export interface InProgressStatus<O extends Operation<string, any>> {
+  operation: O;
+  status: 'in-progress';
+  time: number;
+}
+
+export interface SucceededStatus<O extends Operation<string, any>> {
+  operation: O;
+  status: 'succeeded';
+  time: number;
+}
+
+export interface FailedStatus<O extends Operation<string, any>> {
+  message: string;
+  operation: O;
+  status: 'failed';
+  time: number;
+}
+
```

Comment:
"Operation" is what I call a running request, e.g. a graphql query or mutation. We need to track metadata bout it in order to derive the loading or error states.

Reply:
We can try to come up with a different name than "operation". I used that term from the GraphQL world as a collective term for "query" and "mutation". It could be "AsyncProcess"? "SideEffect"? "Task"?

Reply:
No, it does make sense, thanks for the explanation.